Bundle Vue apps with Vite instead of Webpack

mayakokits opened the pull request:
Replace Webpack with Vite 3 for the Vue apps in e2t, update Vue and some dependencies. Dependencies are bundled with the apps, not used via `campaignion_vue` any more.

mayakokits committed:
e2t: Bundle datasets_app with Vite

- Get rid of campaignion_vue, import dependencies directly
- Update vue to 2.7.7
- Build with Vite 3.2.4	instead of Webpack
- Run unit tests with Vitest
- Use Nightwatch 1.7.13 for end to end testing
- Update some dependencies: axios, element-ui, popper.js, vue-tables-2,
  vuedraggable

mayakokits committed:
e2t: Bundle messages_app with Vite

- Get rid of campaignion_vue, import dependencies directly
- For simplicity, copy DestinationField component and utils
  into this project
- Update vue to 2.7.7
- Build with Vite 3.2.4	instead of Webpack
- Run unit tests with Vitest
- Update some dependencies: axios, element-ui, vuedraggable

mayakokits committed:
e2t datasets_app: Fix el-button not rendering

mayakokits committed:
campaignion_vue: Don't add dependencies for apps that are bundled with them.

mayakokits committed:
campaignion_vue, campaignion_wizard redirects_app: Bump axios to 0.21.1

mayakokits committed:
[github] Run Vue app unit tests in workflow again

torotil commented on campaignion_vue/campaignion_vue.module:
```diff
@@ -58,13 +60,20 @@ function _campaignion_vue_get_strings($language = NULL) {
+  $add_dependencies = FALSE;
   foreach ($js as $path => $options) {
-    if (substr($path, -11) == '.vue.min.js') {
+    if (substr($path, -24) == 'redirects_app.vue.min.js') {
```
I’m not sure where to begin. Actually library dependencies should be handled using hook_library_info(). I don’t think a rewrite makes sense now though. I guess I’d rather go without the optimization of not loading campaignion_vue.min.js on some pages where it is actually not needed.

I guess the problem will go away anyway once the redirect_app uses vite too.

torotil replied:
Ok, talked with Florian. I think it’s better to actually migrate the redirect_app too (and remove the campaignion_vue module) instead of maintaining this in-between state.

torotil commented on campaignion_email_to_target/messages_app/drupal-fixture.js:
```diff
@@ -134,21 +135,19 @@ var Drupal = {
-      delete args['@count'];
-      return Drupal.t(plural.replace('@count', '@count[' + index + ']'), args, options);
+      return Drupal.t(singular, args, options)
+    } else if (index == 1) {
```
Please use [brace-style](https://eslint.org/docs/latest/rules/brace-style) stroustrup for Drupal JS.

mayakokits replied:
Just added a commit to change this, and then I thought a bit... this is not Drupal JS, it's just a partial copy of Drupal JS that is never used in a Drupal context. It's just used by the Vue app in dev and testing mode where the `Drupal` global isn't available.
Also, the Vue app's linter stripped away all the semicolons, which might not fit the Drupal style.
So in my opinion it's cleaner to have this file linted like the rest of the Vue app. This would make my last commit obsolete.

mayakokits committed:
Comply with brace-style "stroustrup"

mayakokits committed:
wizard: Bundle redirects_app with Vite

- Build with Vite 3.2.4 instead of Webpack
- Get rid of campaignion_vue, import dependencies directly
- For simplicity, copy DestinationField component and utils
  into this project
- Update vue to 2.7.7
- Update some dependencies: axios, element-ui, vuedraggable
- Run unit tests with Vitest
- Use Nightwatch 1.7.13 for end to end testing

mayakokits committed:
wizard: lint redirects_app

- update eslint to support import.meta
- update eslint plugins
- update eslint config
- add script to run eslint --fix
- adapt code to comply with the rules

mayakokits committed:
campaignion_vue: clean up after switching Vue apps to Vite

- don't provide a Node module any more
- remove dependencies, they are now bundled with the apps
- adapt Drupal module
- update README

mayakokits committed:
campaignion_vue: update element-ui to match version used by the Vue apps

mayakokits commented:
Now the `redirects_app` is bundled with Vite as well.
We still need the `campaignion_vue` module, as it provides localized strings for the Vue apps that can’t know the user's language at build time. But I cropped it down and removed all that is obsolete.

mayakokits committed:
e2t datasets_app: update Drupal.settings in dev mode

- add settings that were introduced in 3f13594f and c307b02e to be
  available in dev mode
- update README

mayakokits committed:
e2t datasets_app: fix linter errors

mayakokits committed:
e2t datasets_app: build JS

mayakokits committed:
Vue apps: inline element-icons font

mayakokits committed:
Vue apps: fix el-button not rendering

- use the el-button notation throughout the datasets app
- remove invalid type="button" attribute
- add strings to drupal fixtures so they don't throw in dev env
- build all apps

mayakokits committed:
e2t datasets_app: fix vue-tables-2 column headers

mayakokits committed:
e2t datasets_app: fix help icon next to contacts table filter field

mayakokits committed:
e2t messages_app: fix b3166838, re-add bundle

mayakokits committed:
Vue apps: delete obsolete webpack map files
